app/gather/assets/apps/survey/SurveyForm.jsx

```diff
@@ -1073,6 +1096,16 @@ class SurveyForm extends Component {
       url: getSurveysAPIPath({ app: ODK_APP, id: this.state.id, action: 'propagate' })
     })
 
+    if (this.state.odk.xforms.length) {
```

Comment:
No consumer without odk?

Reply:
from gather, yes

Reply:
let me say in another way... no consumers without ODK forms either? In that case, should it mean that if you remove the forms you must remove the consumer subscription too?

Reply:
Removing the forms from a survey does not remove the schema from kernel, and that is what the subscription is kindda tied to. But only a form being present will create the schema, thus, checking for the form before configuring consumers. The subscription is remove when the survey is deleted, cause that propagates to the schema in kernel